dex/elrond_dex_farm/src/lib.rs

```diff
@@ -219,29 +218,41 @@ pub trait Farm:
         }
 
         let farming_token_id = self.farming_token_id().get();
-        let mut farming_token_amount = amount.clone();
+        let mut initial_farming_token_amount = self.rule_of_three(
+            &amount,
+            &farm_attributes.current_farm_amount,
+            &farm_attributes.initial_farming_amount,
+        );
+        reward += self.rule_of_three(
```

Comment:
why not `decrease_reward_reserve` on the compound rewards?

Reply:
the amount was already substracted from reward_reserve upon compoundRewards transaction